### DISCUSSION TITLE
Adding an object tag to individual shapes

### DISCUSSION THREAD

**Discussion by captainkidd5** [notgiven688/jitterphysics2#104] [Ideas]
Would it break anything to add an Object tag to shapes? I'd like to be able to tie data to individual parts of a rigid body instead of just the body as a whole. For example, having a click event on a box which is part of a larger building, but of the same rigid body

**Comment by notgiven688** [comment]
You could just derive your own shape-class:

```
public class TaggedBoxShape : BoxShape
{
    public object? Tag { get; set; }

    public TaggedBoxShape(float xyz) : base(xyz) { }
}
```
... and later...
```
(body.Shapes[0] as TaggedBoxShape).Tag = new object();
```

**Reply by captainkidd5** [reply]
nice! thank you!
